smac_planner/src/a_star.cpp

```diff
@@ -349,6 +366,104 @@ bool AStarAlgorithm<NodeT>::isGoal(NodePtr & node)
   return node == getGoal();
 }
 
+template<>
+AStarAlgorithm<NodeSE2>::NodePtr AStarAlgorithm<NodeSE2>::getAnalyticPath(
+  const NodePtr & node,
+  const NodeGetter & node_getter)
+{
+  std::vector<std::pair<NodePtr, Coordinates>> possible_nodes;
+  const NodePtr & goal = getGoal();
+  ompl::base::ScopedState<> from(node->motion_table.state_space), to(
+    node->motion_table.state_space), s(node->motion_table.state_space);
+  const NodeSE2::Coordinates & node_coords = node->pose;
+  const NodeSE2::Coordinates & goal_coords = goal->pose;
+  from[0] = node_coords.x;
+  from[1] = node_coords.y;
+  from[2] = node_coords.theta * node->motion_table.bin_size;
+  to[0] = goal_coords.x;
+  to[1] = goal_coords.y;
+  to[2] = goal_coords.theta * node->motion_table.bin_size;
+
+  float d = node->motion_table.state_space->distance(from(), to());
+  NodePtr prev(node);
+  // A move of sqrt(2) is guaranteed to be in a new cell
+  constexpr float sqrt_2 = std::sqrt(2.);
+  unsigned int num_pts = std::floor(d / sqrt_2) + 1;
```

Comment:
Reserve the size of the `possible_nodes` vector so that there's no resizing in the function which incurs copies. eg `possible_nodes.reserve(num_pts)'`

Reply:
Done